server/game/cards/01-Core/EpicQuest.js

```diff
@@ -10,14 +10,17 @@ class EpicQuest extends Card {
         });
 
         this.omni({
-            condition: context => context.game.cardsPlayed.filter(card => card.hasHouse('sanctum')).length > 6,
-            effect: 'sacrifice {0} and forge a key at no cost',
-            gameAction: [
-                ability.actions.sacrifice(),
-                ability.actions.forgeKey(context => ({
-                    modifier: -context.player.getCurrentKeyCost()
-                }))
-            ]
+            gameAction: ability.actions.conditional(({
+                effect: 'sacrifice {1} and forge a key at no cost',
```

Comment:
I think this `effect` should be outside the gameAction (as it was before).

Reply:
The problem is that `effect` at ability level always prints the message. But, since this is conditional, I would like to have a conditional message as well.
I renamed the variable to `message`. Maybe it makes more sense.

Reply:
2 problems with that - it will print a message regardless of whether the condition is true or not.  So you either need two messages, or you need your conditional action to be smarter in how it deals with messaging.  If you're keen to implement this functionality, I'd like to see it included in your tests.

Reply:
@jeremylarner : true/false actions are now objects with an internal message.